xls/examples/parameterized/design.x

```diff
@@ -0,0 +1,32 @@
+import xls.examples.parameterized.parameters
+
+type DesignParameters = parameters::DesignParameters;
+type DesignType = parameters::DesignParameters__DesignType;
+
+fn add<INPUT_WIDTH: u32, OUTPUT_WIDTH: u32>(x: uN[INPUT_WIDTH], y: uN[INPUT_WIDTH]) -> uN[OUTPUT_WIDTH] {
+    let lhs = x as uN[OUTPUT_WIDTH];
+    let rhs = y as uN[OUTPUT_WIDTH];
+    lhs + rhs
+}
+
+fn sub<INPUT_WIDTH: u32, OUTPUT_WIDTH: u32>(x: uN[INPUT_WIDTH], y: uN[INPUT_WIDTH]) -> uN[OUTPUT_WIDTH] {
+    let lhs = x as uN[OUTPUT_WIDTH];
+    let rhs = y as uN[OUTPUT_WIDTH];
+    lhs - rhs
+}
+
+fn top(x: u64, y: u64) -> u64 {
+  let variant: DesignParameters = parameters::parameters.variant[0];
+  let variant_type: DesignType = variant.design_type;
+  let variant_input_width: u32 = variant.input_width;
+  let variant_output_width: u32 = variant.output_width;
+  let arg_x = x as uN[variant_input_width];
+  let arg_y = y as uN[variant_input_width];
+
+  let result = if (variant_type == DesignType::DESIGN_TYPE_ADD) {
```

Comment:
maybe we should stick to paramertizing parameter of a single function (I would argue that switching between function is less common and not necessarly what `proto_to_dslx` seems designed for)

Reply:
Here I just wanted to show a different kind of parameter. Something different than I/O width, something that would change the behaviour of the whole module. I'm not that much experienced in DSLX and that seemed like the easiest way to achieve that effect. We should probably change that parameter to something else if we want a real-life example or leave it like that in order to keep it simple and only showcase the parameterization mechanism on minimal example.

Reply:
It'd focus on a minimal example for now, dslx paremetric behavior is not yet very advanced and is not really meant to swap behavior afaik.

Reply:
Alright, so in order to keep this example minimal I will remove `design_type` parameter and leave only IO width.